core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -305,7 +325,7 @@ class ReplicaManager(val config: KafkaConfig,
                     responseCallback: Map[TopicPartition, PartitionResponse] => Unit) {
 
     if (isValidRequiredAcks(requiredAcks)) {
-      val sTime = time.milliseconds
+      val sTime = if (logger.isDebugEnabled) time.milliseconds else 0
```

Comment:
Do you mean `debug("Produce to local log in %d ms".format(time.milliseconds - sTime))`? I thought this `time.milliseconds` will only be evaluated if debug level is true, no?